lib/puppet/provider/ec2_instance/v2.rb

```diff
@@ -63,6 +64,7 @@ def self.instance_to_hash(region, instance)
       instance_type: instance.instance_type,
       image_id: instance.image_id,
       instance_id: instance.instance_id,
+      id: instance.instance_id,
```

Comment:
instance_id is exposed in the type, so you can access it via `puppet resource` if you need to. id is an implementation detail, it's used for shared methods (like the `tags=` method). So anything that want's to use `tags=` has to expose id in the `property_hash`.